java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/maplegend/SvgMapLegendGenerator.java

```diff
@@ -0,0 +1,346 @@
+package net.osmand.maplegend;
+import net.osmand.PlatformUtil;
+import net.osmand.render.RendererRulesStorageAmenityFetcher;
+import net.osmand.util.Algorithms;
+import org.apache.commons.io.FileUtils;
+import org.xmlpull.v1.XmlPullParser;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+
+// Launches from MainUtilities with CLI Arguments: generate-maplegend-svg
+// Also you need to add environment variables like this:
+// web.repo.dir=/Users/nnngrach/OsmAnd/web;resources.repo.dir=/Users/nnngrach/OsmAnd/resources
+
+public class SvgMapLegendGenerator {
+
+    static String configPath = System.getenv("resources.repo.dir") + "/rendering_styles/map-legend/map-legend.xml";
+    static String resultIconsFolderPath = System.getenv("web.repo.dir") + "/main/static/img/legend/osmand/";
+    static String resultComponentsPath = System.getenv("web.repo.dir") + "/main/src/components/docs/autogenerated/";
+
+    static String shieldsFolder = System.getenv("resources.repo.dir") + "/icons/svg/shields/";
+    static String iconsFolder = System.getenv("resources.repo.dir") + "/rendering_styles/style-icons/poi-icons-svg/";
+
+    static int defaultZoomLevel = 19; //Most of the icons on this zoom are visible
+    static int canvasWidth = 300;
+    static int canvasHeight = 35;
+    static double shieldSize = 30;
+    static double iconSize = 18;
+
+    public static void main(String[] args) throws Exception{
+        System.out.println("\n\n\n");
+        System.out.println("OK: SvgMapLegendGenerator - Start script");
+        try {
+
+            if (Algorithms.isEmpty(System.getenv("web.repo.dir")) || Algorithms.isEmpty(System.getenv("resources.repo.dir"))) {
```

Comment:
should be just 1 variable "repo.dir"
args[0] - repo.dir 
args[1] - rendering name (same as config.xml name in resources)